lib/ExecutionContext.js

```diff
@@ -76,8 +78,23 @@ class ExecutionContext {
       return this._objectHandleFactory(remoteObject);
     }
 
+    if (typeof pageFunction !== 'function')
+      throw new Error('The following is not a function: ' + pageFunction);
+
+    const originalFunctionDeclaration = pageFunction.toString();
+    // We can't add the sourceURL comment after the function body because then it becomes an invalid function declaration.
+    // The comment should be added inside the function body instead.
+    // For a function enclosed in curly braces, it's as easy as adding the line before the closing curly brace.
+    // For an arrow function with a single expression and no braces, we should first add the braces and a return statement.
+    const functionDeclarationInBraces = originalFunctionDeclaration[originalFunctionDeclaration.length - 1] === '}' ?
+      originalFunctionDeclaration :
+      (() => {
+        const indexAfterArrow = originalFunctionDeclaration.indexOf('=>') + 2;
+        return originalFunctionDeclaration.slice(0, indexAfterArrow) + '{ return ' + originalFunctionDeclaration.slice(indexAfterArrow) + '}';
+      })();
+    const declarationWithSourceURL = functionDeclarationInBraces.slice(0, functionDeclarationInBraces.length - 1) + '\n' + suffix + '\n}';
     const { exceptionDetails, result: remoteObject } = await this._client.send('Runtime.callFunctionOn', {
-      functionDeclaration: pageFunction.toString(),
+      functionDeclaration: declarationWithSourceURL,
```

Comment:
instead of computing `declarationWithSourceURL` in this fancy way, you [should be able](https://cs.chromium.org/chromium/src/v8/src/inspector/v8-runtime-agent-impl.cc?type=cs&q=callFunctionOn&sq=package:chromium&g=0&l=146) to just pass sourceURL with a new line in the end:

```js
  // ...
  fuctionDeclaration: pageFunction.toString() + '\n//# sourceURL=somename.js\n',
  // ...
```
Trailing `\n` should do the trick.